File: nha-trang-bus-tour/public/src/main/java/com/grouppro/nhatrangbustour/Entity/Ticket.java
package com.grouppro.nhatrangbustour.Entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Nationalized;

import java.io.Serializable;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "Ticket")
@Table(name = "Ticket")
public class Ticket implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "TicketID", updatable = false)
    private Long TicketID;
    @Nationalized
    @Column(
            name = "PassengerName",
            nullable = false,
            length = 100
    )
    private String PassengerName;
    @Nationalized
    @Column(
            name = "PassengerPhone",
            nullable = false,
            length = 100
    )
    private String PassengerPhone;
    @Nationalized
    @Column(
            name = "PassengerEmail",
            nullable = false,
            length = 100
    )
    private String PassengerEmail;
    @Nationalized
    @Column(
            name = "Feedback",
            nullable = false,
            length = 100
    )
    private String Feedback;
    @ManyToOne(fetch = FetchType.LAZY)
    @JsonManagedReference
    @JoinColumn(name = "TripID")
    private Trip Trip;
    @ManyToOne(fetch = FetchType.LAZY)
    @JsonManagedReference
    @JoinColumn(name = "OrderID")
    private Order Order;
    @ManyToOne(fetch = FetchType.LAZY)
    @JsonManagedReference
    @JoinColumn(name = "ServiceID")
    private Service Service;
    @ManyToOne(fetch = FetchType.LAZY)
    @JsonManagedReference
    @JoinColumn(name = "TicketTypeID")
    private TicketType TicketType;
}

File: swd392/lib/model/ticket.dart
class Ticket {
  int? ticketId;
  String? passengerName;
  String? passengerPhone;
  String? passengerEmail;
  String? feedback;

  Ticket(
      {this.ticketId,
        this.passengerName,
        this.passengerPhone,
        this.passengerEmail,
        this.feedback});

  Ticket.fromJson(Map<String, dynamic> json) {
    ticketId = json['ticketId'];
    passengerName = json['passengerName'];
    passengerPhone = json['passengerPhone'];
    passengerEmail = json['passengerEmail'];
    feedback = json['feedback'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['ticketId'] = this.ticketId;
    data['passengerName'] = this.passengerName;
    data['passengerPhone'] = this.passengerPhone;
    data['passengerEmail'] = this.passengerEmail;
    data['feedback'] = this.feedback;
    return data;
  }
}